source/game.gamescriptexpression.bmx

```diff
@@ -4,37 +4,627 @@ Import "game.world.worldtime.bmx"
+		Case "broadcastlimitmax"       Return New SToken( TK_NUMBER, licence.GetBroadcastLimitMax(), params.GetToken(0) )
+		Case "broadcastlimit"          Return New SToken( TK_NUMBER, licence.GetBroadcastLimit(), params.GetToken(0) )
+		Case "hasbroadcastlimit"       Return New SToken( TK_BOOLEAN, licence.HasBroadcastLimit(), params.GetToken(0) )
+		Case "episodenumber"           Return New SToken( TK_NUMBER, licence.GetEpisodeNumber(), params.GetToken(0) )
+		Case "episodecount"            Return New SToken( TK_NUMBER, licence.GetEpisodeCount(), params.GetToken(0) )
+		'Case "isavailable"             Return New SToken( TK_BOOLEAN, licence.isAvailable(), params.GetToken(0) )
+		'Case "isreleased"              Return New SToken( TK_BOOLEAN, licence.isReleased(), params.GetToken(0) )
+		'Case "isplanned"               Return New SToken( TK_BOOLEAN, licence.isPlanned(), params.GetToken(0) )
+		'Case "isprogrammeplanned"      Return New SToken( TK_BOOLEAN, licence.isProgrammePlanned(), params.GetToken(0) )
+		'Case "istrailerplanned"        Return New SToken( TK_BOOLEAN, licence.isTrailerPlanned(), params.GetToken(0) )
+		'Case "isnewbroadcastpossible"  Return New SToken( TK_TEXT, licence.IsNewBroadcastPossible(), params.GetToken(0) )
+		Case "genre"                   Return New SToken( TK_TEXT, licence.GetGenre(), params.GetToken(0) )
+		Case "genrestring"             Return New SToken( TK_TEXT, licence.GetGenreString(), params.GetToken(0) )
+		Case "genresline"              Return New SToken( TK_TEXT, licence.GetGenresLine(), params.GetToken(0) )
+		Case "hasdataflag"             Return New SToken( TK_BOOLEAN, licence.HasDataFlag(Int(params.GetToken(3).valueLong)), params.GetToken(0) )
+		Case "hasbroadcastflag"        Return New SToken( TK_BOOLEAN, licence.HasBroadcastFlag(Int(params.GetToken(3).valueLong)), params.GetToken(0) )
+		Case "hasflag"                 Return New SToken( TK_BOOLEAN, licence.HasFlag(Int(params.GetToken(3).valueLong)), params.GetToken(0) )
+		Case "quality"                 Return New SToken( TK_NUMBER, licence.GetQuality(), params.GetToken(0) )
+		Case "speed"                   Return New SToken( TK_NUMBER, licence.GetSpeed(), params.GetToken(0) )
+		Case "review"                  Return New SToken( TK_NUMBER, licence.GetReview(), params.GetToken(0) )
+		Case "outcome"                 Return New SToken( TK_NUMBER, licence.GetOutcome(), params.GetToken(0) )
+		Case "outcometv"               Return New SToken( TK_NUMBER, licence.GetOutcomeTV(), params.GetToken(0) )
+		Case "blocks"                  Return New SToken( TK_NUMBER, licence.GetBlocks(), params.GetToken(0) )
+		Case "relativetopicality"      Return New SToken( TK_NUMBER, licence.GetRelativeTopicality(), params.GetToken(0) )
+		Case "topicality"              Return New SToken( TK_NUMBER, licence.GetTopicality(), params.GetToken(0) )
+		Case "maxtopicality"           Return New SToken( TK_NUMBER, licence.GetMaxTopicality(), params.GetToken(0) )
+		Case "cast"
```

Comment:
Wie ist die Performance von case? Die meisten Attribute, die vor cast stehen, werden kaum benutzt werden (zumindest in der Datenbank). Den Teil nach oben ziehen, für schnellere Matches?

cast gibt es bei Scripten und Programmen, in beiden Fällen geht die Auflösung an Person... Code nicht doppeln sondern rausziehen (ggf. direkt auf Personen-Expressions delegieren, dann lassen sich auch andere Personenattibute verwenden)

Reply:
Es wird natuerlich ein "if ... elseif ... elseif ... elseif...". Inwieweit das am Ende aber irgend eine "spuerbare" Rolle spielt, bezweifle ich (also diese Stelle vs "der ganze, sicher langsamere, Rest"). Kann man aber gerne nach Wunsch sortieren (im Nachgang?).

Schlaegt halt in die gleiche Kerbe wie `${.worldtime_year}` (verlangsamt generelle "main key"-lookups durch die Anzahl) vs `{.worldtime:"year"}` (fuer "year" wird einmalig der hash berechnet und dann mit den hashes in den cases _innerhalb_ des "worldtime-handlers" verglichen).

Reply:
Gern im Nachgang nach wahrscheinlicher Verwendung sortieren. Bei vielen von den Feldern könnte ich mir vorstellen, dass sie praktisch nie für die Datenbank interessant werden.